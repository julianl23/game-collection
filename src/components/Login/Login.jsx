import axios from 'axios';
import React, { Component } from 'react';
import cn from 'classnames';

class Login extends Component {
  submitLogin(e) {
    e.preventDefault();
    const form = e.target;
    const loginData = {
      username: form.username.value,
      password: form.password.value
    };

    axios.post('/api/login', loginData)
      .then((res) => {
        console.log('logged in', res.data.user);
      })
      .catch((res) => {
        console.log('login failed', res);
      });
  }

  render() {
    return(
      <div className={cn('cp-login')}>
        <h1>Log In</h1>

        <form onSubmit={this.submitLogin}>
          <div className="form-group">
            <label htmlFor="username">Username</label>
            <input type="text" id="username" placeholder="Username" />
          </div>
          
          <div className="form-group">
            <label htmlFor="password">Password</label>
            <input type="password" id="password" placeholder="Password" />
          </div>

          <button type="submit">Log in</button>
        </form>
      </div>
    );
  }
}

export default Login;