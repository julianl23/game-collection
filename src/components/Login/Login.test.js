import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jsdom-global/register';
import Login from './Login';

it('renders without crashing', () => {
  // const wrapper = mount(<Login />);
  // expect(wrapper.find('.cp-login').length).toBe(1);

  const tree = renderer.create(
    <Login />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
