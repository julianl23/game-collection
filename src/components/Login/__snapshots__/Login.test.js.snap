// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="cp-login"
>
  <h1>
    Log In
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        id="username"
        placeholder="Username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Log in
    </button>
  </form>
</div>
`;
